<a href="https://colab.research.google.com/github/dsb4k8/DS-5530_Assignment_2_3_Combined/blob/main/Assignment_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1 align="center">Assignment2</h1>


### Data link Q1:  https://app.box.com/s/jm6pw202asu4xd3uypwtry2rqk691y1i
1) >The provided data (link above) contains various details and attributes associated with used cars. The target variable, which is the central focus of analysis, is the price of the used cars, and it is measured in lakhs. The data in this dataset is tabular, with rows and columns, where each row represents a specific used car listing, and each column represents a particular attribute or feature of these cars.  Features are Make and model of the car, Location or city of sale, Year of manufacture, Mileage, Odometer (kilometers driven), Fuel type (petrol or diesel), Transmission type (manual or automatic), Number of owners, Engine displacement, Engine horsepower, Number of seats, and Price when the car was new.

In [5]:
import numpy as np
import pandas as pd

### Setup: Getting Data from Box

In [2]:
# import requests
# import os
#
# url = "https://app.box.com/s/jm6pw202asu4xd3uypwtry2rqk691y1i"
# target_dir = "Assignment_2/data_raw"
# filename = "train.csv"
#
# # Ensure the directory exists
# os.makedirs(target_dir, exist_ok=True)
#
# # Download and save the file
# response = requests.get(url)
# with open(os.path.join(target_dir, filename), 'wb') as f:
#     f.write(response.content)

### Setup: Preprocessing

In [7]:
df = pd.read_csv('Assignment_2/data_raw/train.csv')
print(df.head())

   Unnamed: 0                              Name    Location  Year  \
0           1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1           2                      Honda Jazz V     Chennai  2011   
2           3                 Maruti Ertiga VDI     Chennai  2012   
3           4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4           6            Nissan Micra Diesel XV      Jaipur  2013   

   Kilometers_Driven Fuel_Type Transmission Owner_Type     Mileage   Engine  \
0              41000    Diesel       Manual      First  19.67 kmpl  1582 CC   
1              46000    Petrol       Manual      First    13 km/kg  1199 CC   
2              87000    Diesel       Manual      First  20.77 kmpl  1248 CC   
3              40670    Diesel    Automatic     Second   15.2 kmpl  1968 CC   
4              86999    Diesel       Manual      First  23.08 kmpl  1461 CC   

       Power  Seats  New_Price  Price  
0  126.2 bhp    5.0        NaN  12.50  
1   88.7 bhp    5.0  8.61 Lakh

## EDA

In [10]:
# count of missing values
missing_counts = df.isnull().sum()

# calculate percentages
missing_pct = (df.isnull().sum() / len(df)) * 100

# 3. Combine for a clear report
missing_report = pd.DataFrame({
    'Missing Count': missing_counts,
    'Percentage (%)': missing_pct
}).sort_values(by='Percentage (%)', ascending=False)

print(missing_report)


                   Missing Count  Percentage (%)
New_Price                   5032       86.061228
Seats                         38        0.649906
Engine                        36        0.615700
Power                         36        0.615700
Mileage                        2        0.034206
Unnamed: 0                     0        0.000000
Name                           0        0.000000
Location                       0        0.000000
Year                           0        0.000000
Kilometers_Driven              0        0.000000
Fuel_Type                      0        0.000000
Transmission                   0        0.000000
Owner_Type                     0        0.000000
Price                          0        0.000000


Findings:

1) New_Price column value is missing in >86% of records. Filling in such a large percentage of the column with averages, modes, etc from such a small sample would introduce alot of bias and noise. On the other hand, dropping 86% of the rows  leaves almost no data to analyze. For these reasons I think this column should be dropped all together

2) Seats, Engine, Power, and Mileage on the other hand have over 99% non-null values. I think these records can be safely dropped. This allows the model to learn from real data without significantly reducing the size of the original dataset.